File: backend/voting-app/src/main/java/com/tu/votingapp/entities/DocumentEntity.java
package com.tu.votingapp.entities;

import jakarta.persistence.*;
import lombok.*;

import java.sql.Date;

@Entity(name = "documents")
@Table(name = "documents")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode(callSuper = false)
@ToString()
public class DocumentEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id", nullable = false, updatable = false)
    private Long id;

    @NonNull
    @Column(name = "document_number", nullable = false, unique = true, updatable = false)
    private String number;

    @NonNull
    @Column(name = "valid_from", nullable = false, updatable = false)
    private Date validFrom;

    @NonNull
    @Column(name = "valid_to", nullable = false, updatable = false)
    private Date validTo;

    @NonNull
    @Column(name = "document_issuer", nullable = false, updatable = false)
    private String issuer;

    @NonNull
    @Column(name = "gender", nullable = false, updatable = false)
    private Integer gender;

    @NonNull
    @Column(name = "date_of_birth", nullable = false, updatable = false)
    private Date dateOfBirth;

    @NonNull
    @Column(name = "permanent_address", nullable = false, updatable = false)
    private String permanentAddress;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private UserEntity user;

}

File: frontend/android/app/src/main/java/com/example/android/entity/DocumentEntity.kt
package com.example.android.entity

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey
import com.example.android.db.orm.Column
import com.example.android.db.orm.Table

@Entity(
    tableName = "Document",
    foreignKeys = [
        ForeignKey(
            entity = UserEntity::class,
            parentColumns = ["id"],
            childColumns = ["userId"],
            onDelete = ForeignKey.CASCADE
        )
    ]
)
data class DocumentEntity(
    @PrimaryKey(autoGenerate = true)
    @ColumnInfo(name = "id")
    val id: Long = 0,

    @ColumnInfo(name = "number")
    val number: String,

    @ColumnInfo(name = "validFrom")
    val validFrom: String, // Stored as ISO 8601 string: "yyyy-MM-dd"

    @ColumnInfo(name = "validTo")
    val validTo: String,

    @ColumnInfo(name = "issuer")
    val issuer: String,

    @ColumnInfo(name = "gender")
    val gender: Int,

    @ColumnInfo(name = "dateOfBirth")
    val dateOfBirth: String,

    @ColumnInfo(name = "permanentAddress")
    val permanentAddress: String,

    @ColumnInfo(name = "userId")
    val userId: Long // foreign key to UserEntity
)